cli/lib/pm2-manager.ts

```diff
@@ -113,61 +129,84 @@ function getProxyProcessEnv() {
 		env.ELECTRON_RUN_AS_NODE = '1';
 	}
 
-	return env;
+	return startProcess( PROXY_PROCESS_NAME, proxyDaemonPath, env );
 }
 
-/**
- * Start the proxy server via PM2
- * This launches the proxy-daemon.js script which runs the proxy servers
- */
-export async function startProxyProcess(): Promise< ProcessDescription > {
-	const proxyDaemonPath = path.resolve( __dirname, 'proxy-daemon.js' );
+export async function isProxyProcessRunning(): Promise< boolean > {
+	return isProcessRunning( PROXY_PROCESS_NAME );
+}
+
+export async function isProcessRunning( processName: string ): Promise< boolean > {
+	try {
+		if ( ! isConnected ) {
+			return false;
+		}
+
+		const processes = await listProcesses();
+		return processes.some( ( p ) => p.name === processName && p.status === PM2_STATUS_ONLINE );
+	} catch ( error ) {
+		console.error( `Error checking if process ${ processName } is running:`, error );
+		return false;
+	}
+}
+
+export async function startProcess(
+	processName: string,
+	scriptPath: string,
+	env?: Record< string, string >
+): Promise< ProcessDescription > {
 	return new Promise( ( resolve, reject ) => {
 		const processConfig: StartOptions = {
-			name: PROXY_PROCESS_NAME,
+			name: processName,
 			interpreter: process.execPath,
-			script: proxyDaemonPath,
+			script: scriptPath,
 			exec_mode: 'fork',
 			autorestart: true,
 			max_restarts: 5,
```

Comment:
```suggestion
autorestart: false,
```
Let's change this. If a `wordpress-server-child` process fails to start the Playground server, the process should die. However, this config will cause pm2 to restart the process and leave it hanging indefinitely, since it needs the `start-server` message to actually initiate Playground.